src/kw_include.sh

```diff
@@ -0,0 +1,14 @@
+# Used for file sourcing,
+# accepts as argument the path for the file to be sourced
+function include() {
+  local filename="$1"
+  local varname=${filename%.sh}
+  varname=${varname^^}_IMPORTED
+
+  if [ -n "${!varname}" ]; then
```

Comment:
@rodrigosiqueira notice the `!` indicating an indirect expansion. If the file is not sourced, the variable whose name is defined by `$varname` will not be defined. So `${!varname}` will expand to `""` and thus this test will not be true.

Reply:
Hi @alan-barzilay and @joaoseckler,

> notice the ! indicating an indirect expansion. If the file is not sourced, the variable whose name is defined by $varname will not be defined. So ${!varname} will expand to "" and thus this test will not be true.

Hmmm... interesting...
I checked in the documentation, and the only reference for `!` expansion that I found was:

> ${!prefix*}
> ${!prefix@}
> Expands to the names of variables whose names begin with prefix, separated by the first character of the IFS special variable. When ‘@’ is used and the expansion appears within double quotes, each variable name expands to a separate word.
> 
> ${!name[@]}
> ${!name[*]}
> If name is an array variable, expands to the list of array indices (keys) assigned in name. If name is not an array, expands to 0 if name is set and null otherwise. When ‘@’ is used and the expansion appears within double quotes, each key expands to a separate word.
[[1](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html)]

Notice that the above references are different from the ones that you used here. Did I miss something? Could you point me to a better reference?

Additionally, with this code, I'm getting this error message: `invalid variable name`.

Furthermore, we will always get this condition set to true due to the way that you declared `varname`:

```
varname=${varname^^}_IMPORTED
```

It does not matter the result from `${varname^^}`, `varname` will always have `_IMPORTED`. 

Finally, when I tested this PR, kw was totally broke; I could not see any feature working as expected. Install this PR and try a simple command like `kw c PATH/` OR `kw m PATH/`.

Reply:
I have found a better way of doing this, newer versions of bash have a -v flag that checks if a variable is set

Reply:
Please, read all my previous comments carefully. Your new patch still has some of the problems that I pointed before.

Reply:
About indirect expansions, the documentation can be found (here)[https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion]: 

> If the first character of parameter is an exclamation point (!), and parameter is not a nameref, it introduces a level of indirection. Bash uses the value formed by expanding the rest of parameter as the new parameter; this is then expanded and that value is used in the rest of the expansion, rather than the expansion of the original parameter. This is known as indirect expansion.

By the way, [this feature is in kw's codebase](https://github.com/kworkflow/kworkflow/blob/7b7b8e44d5f65e7f29188958423b044edfafebf7/run_tests.sh#L55). But @alan-barzilay found a more elegant solution  using `-v` inside the double brackets.